### ISSUE TITLE
messenger_testing_utils.go pollutes production scope

### ISSUE THREAD

**Issue by osmaczko** [status-im/status-go#4597]
Files without `_test.go` suffix are compiled and included into the package when building the production binary.

See https://github.com/status-im/status-go/issues/4594#issuecomment-1900283194_

**Comment by osmaczko** [comment]
FYI: @igor-sirotin

**Comment by igor-sirotin** [comment]
> Files without `_test.go` suffix are compiled and included into the package when building the production binary.

Yes, but as far as I remember I couldn't use these functions when they were in `_test.go` file.

**Comment by osmaczko** [comment]
> Yes, but as far as I remember I couldn't use these functions when they were in _test.go file.

Correct, you cannot use test utilities defined in one package within the *_test.go files of another package. However, `messenger_testing_utils.go` might not be the best approach as stated in the issue.
